internal/storage/activities/delete_amss_test.go

```diff
@@ -0,0 +1,339 @@
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "request deletion: response code: 500",
+		},
+		{
+			name:    "Fails requesting deletion (decode error)",
+			approve: true,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				switch r.URL.Path {
+				case "/api/v2/file/" + aipUUID + "/":
+					handleAIPInfo(w, r, "DEL_REQ")
+				case "/api/v2/file/" + aipUUID + "/delete_aip/":
+					w.Header().Set("Content-Type", "application/json")
+					_, err := w.Write([]byte("{"))
+					assert.NilError(t, err)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "request deletion: failed to decode response body",
+		},
+		{
+			name:    "Fails approving deletion (HTTP error)",
+			approve: true,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				switch r.URL.Path {
+				case "/api/v2/file/" + aipUUID + "/":
+					handleAIPInfo(w, r, "DEL_REQ")
+				case "/api/v2/file/" + aipUUID + "/delete_aip/":
+					handleRequest(w, r)
+				case "/api/v2/file/" + aipUUID + "/review_aip_deletion/":
+					http.Error(w, "", http.StatusInternalServerError)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "approve deletion: response code: 500",
+		},
+		{
+			name:    "Fails approving deletion (decode error)",
+			approve: true,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				switch r.URL.Path {
+				case "/api/v2/file/" + aipUUID + "/":
+					handleAIPInfo(w, r, "DEL_REQ")
+				case "/api/v2/file/" + aipUUID + "/delete_aip/":
+					handleRequest(w, r)
+				case "/api/v2/file/" + aipUUID + "/review_aip_deletion/":
+					w.Header().Set("Content-Type", "application/json")
+					_, err := w.Write([]byte("{"))
+					assert.NilError(t, err)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "approve deletion: failed to decode response body",
+		},
+		{
+			name:    "Fails approving deletion (response error_message)",
+			approve: true,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				switch r.URL.Path {
+				case "/api/v2/file/" + aipUUID + "/":
+					handleAIPInfo(w, r, "DEL_REQ")
+				case "/api/v2/file/" + aipUUID + "/delete_aip/":
+					handleRequest(w, r)
+				case "/api/v2/file/" + aipUUID + "/review_aip_deletion/":
+					w.Header().Set("Content-Type", "application/json")
+					err := json.NewEncoder(w).Encode(map[string]any{"error_message": "error message"})
+					assert.NilError(t, err)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "approve deletion: error message",
+		},
+		{
+			name: "Deletes AIP polling for status (approved)",
+			handler: func() http.HandlerFunc {
+				var requested, polled bool
+				return func(w http.ResponseWriter, r *http.Request) {
+					switch r.URL.Path {
+					case "/api/v2/file/" + aipUUID + "/":
+						if !requested {
+							handleAIPInfo(w, r, "DEL_REQ")
```

Comment:
Shouldn't the server return a status other than "DEL_REQ" before the deletion is requested?